### PR TITLE
bump k8s.io dependencies to 1.17.2

### PR DESCRIPTION
**Pull request by MartinForReal**
move k8s.io/kubernetes/pkg/ssh to k8s.io/node-problem-detector/test/e2e/lib/ssh/k8sssh because ssh package is removed from k8s.io/kubernetes
remove k8s.io/kubernetes from dependencies
bump k8s.io dependencies to 1.17.2

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,89 +3,103 @@ module k8s.io/node-problem-detector
...
+	github.com/stretchr/testify v1.8.1
+	go.opencensus.io v0.24.0
+	golang.org/x/crypto v0.1.0
+	golang.org/x/oauth2 v0.7.0
+	golang.org/x/sys v0.8.0
+	google.golang.org/api v0.114.0
+	k8s.io/api v0.17.2
+	k8s.io/apimachinery v0.17.2
 	k8s.io/client-go v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible
-	k8s.io/kubernetes v1.14.6
```
Comment (mmiranda96) [inline go.mod:33]: Why are we removing kubernetes as a dep?

**Review by mmiranda96 — COMMENTED** [review]
Thanks for this PR! I was working on something similar, but you beat me to it.

Could you upgrade to a later version of k8s.io? Perhaps 1.27, the latest available one.

File: go.mod (hunk 1 above)
Comment (MartinForReal) [reply go.mod:33]: In the later version, we need to add a lot of replace statements. It is much simpler if there are fewer dependencies.

**Comment by MartinForReal** [comment]
Thank you for the review! 
In k8s 1.19/1.18 release context. Context is introduced into the function parameters. I think we need to do the same to the NPD. And there will be less compatibility issues if the version gap is not that huge(api/behavior mismatch, package missing, function refactor and etc.). So, my plan would be upgrading to 1.19(adding context) and then upgrading to 1.24（to the oldest supported version ）How do you think?

**Comment by yaron-idan** [comment]
Hey @MartinForReal and @mmiranda96 - I was also looking into upgrading k8s dependencies to address #742, and opened this PR to perform the upgrade - #761. Do you think we can merge it instead to get closer to the current k8s version?

**Comment by MartinForReal** [comment]
@mmiranda96 Which choice do you prefer? upgrade to 1.25 directly and put all of changes mentioned above into one pr?

**Comment by mmiranda96** [comment]
Apologies for the delay. Since these upgrades result in very heavy PRs, I would prefer to review several smaller PRs (easier to review + Github page doesn't crash that often).

**Comment by MartinForReal** [comment]
Glad to know that. 

> Hey @MartinForReal and @mmiranda96 - I was also looking into upgrading k8s dependencies to address #742, and opened this PR to perform the upgrade - #761. Do you think we can merge it instead to get closer to the current k8s version?
I would suggest putting code changes into separate pr. @yaron-idan How do you think?

**Comment by yaron-idan** [comment]
@MartinForReal @mmiranda96 - so the suggestion is to merge this PR, and then merge the changes into #761 so it would include fewer changes? If so, I think it's a good idea.

**Comment by MartinForReal** [comment]
@mmiranda96 Could you please review this pr?

File: test/e2e/lib/ssh/k8sssh/ssh.go [hunk 2]
```diff
@@ -36,33 +38,44 @@ import (
...
 
 	utilnet "k8s.io/apimachinery/pkg/util/net"
 	"k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
 	"k8s.io/klog"
 )
 
+/*
```
Comment (mmiranda96) [inline test/e2e/lib/ssh/k8sssh/ssh.go:51]: Consider that the new file is under k8s.io/node-problem-detector/test/e2e/lib/ssh/k8sssh/. If the package name is `ssh`, then the code that imports it need to reference it as `k8s.io/node-problem-detector/test/e2e/lib/ssh/k8sssh/ssh`. This is quite redundant.

I think a cleaner solution is to move vendor/k8s.io/kubernetes/pkg/ssh.go to test/e2e/lib/ssh/lib.go and vendor/k8s.io/kubernetes/pkg/lib_test.go. You can also add a note explaining where and why we moved it.

File: go.mod (hunk 1 above)
Comment (mmiranda96) [reply go.mod:33]: I understand, but not sure if this is correct. Let's keep it on the same version for now, we can send a follow-up PR just to update it.

**Review by mmiranda96 — CHANGES_REQUESTED** [review]
I left a couple of comments, please take a look. I'll continue reviewing the vendor/ directory, please bear with me as a PR this size will take me some time to review.

File: go.mod (hunk 1 above)
Comment (MartinForReal) [reply go.mod:33]: ssh packageis the only package

File: pkg/systemlogmonitor/logwatchers/filelog/translator.go [hunk 3]
```diff
@@ -5,7 +5,7 @@ Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-    http://www.apache.org/licenses/LICENSE-2.0
+	http://www.apache.org/licenses/LICENSE-2.0
```
[the remark this answers is not in the document]
Comment (MartinForReal) [reply pkg/systemlogmonitor/logwatchers/filelog/translator.go]: files are reverted.

**Comment by MartinForReal** [comment]
@mmiranda96 comment is addressed in new commit. Could you please kindly review this pr? Thanks!

File: pkg/systemstatsmonitor/net_collector_test.go [hunk 4]
```diff
@@ -12,22 +12,22 @@ import (
...
-	string(metrics.NetDevRxMulticast):  ssmtypes.MetricConfig{DisplayName: string(metrics.NetDevRxMulticast)},
-	string(metrics.NetDevTxBytes):      ssmtypes.MetricConfig{DisplayName: string(metrics.NetDevTxBytes)},
-	string(metrics.NetDevTxPackets):    ssmtypes.MetricConfig{DisplayName: string(metrics.NetDevTxPackets)},
-	string(metrics.NetDevTxErrors):     ssmtypes.MetricConfig{DisplayName: string(metrics.NetDevTxErrors)},
-	string(metrics.NetDevTxDropped):    ssmtypes.MetricConfig{DisplayName: string(metrics.NetDevTxDropped)},
-	string(metrics.NetDevTxFifo):       ssmtypes.MetricConfig{DisplayName: string(metrics.NetDevTxFifo)},
-	string(metrics.NetDevTxCollisions): ssmtypes.MetricConfig{DisplayName: string(metrics.NetDevTxCollisions)},
-	string(metrics.NetDevTxCarrier):    ssmtypes.MetricConfig{DisplayName: string(metrics.NetDevTxCarrier)},
-	string(metrics.NetDevTxCompressed): ssmtypes.MetricConfig{DisplayName: string(metrics.NetDevTxCompressed)},
+	string(metrics.NetDevRxBytes):      {DisplayName: string(metrics.NetDevRxBytes)},
```
[the remark this answers is not in the document]
Comment (MartinForReal) [reply pkg/systemstatsmonitor/net_collector_test.go]: the change is reverted.

**Review by vteratipally — APPROVED** [review]

**Pull request merged**